### PR TITLE
add validation on finished projects; add research content type enum

### PR DESCRIPTION
**Pull request by andrew-polyansky**

### CODE REVIEW

File: pallets/deip/src/lib.rs [hunk 1]
```diff
@@ -14,6 +14,37 @@ use sp_core::{ H160 };
 /// A maximum number of Domains. When domains reaches this number, no new domains can be added.
 pub const MAX_DOMAINS: u32 = 100;
 
+#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq)]
+enum ResearchContentType {
```
Comment (egortsaryk9) [inline pallets/deip/src/lib.rs:18]: Let's rename this enum to "ProjectContentType" according to new conventions

File: pallets/deip/src/lib.rs [hunk 2]
```diff
@@ -14,6 +14,37 @@ use sp_core::{ H160 };
...
+    Announcemen,
+    FinalResul,
+    MilestoneArticl,
+    MilestoneBoo,
+    MilestoneChapte,
+    MilestoneCod,
+    MilestoneConferencePape,
+    MilestoneCoverPag,
```
Comment (egortsaryk9) [inline pallets/deip/src/lib.rs:26]: I'm not sure, why there are missing letters at the end of some words... Is it correct?

File: pallets/deip/src/lib.rs (hunk 1 above)
Comment (andrew-polyansky) [reply pallets/deip/src/lib.rs]: Yep, thanks for pointing this

File: pallets/deip/src/lib.rs (hunk 2 above)
Comment (andrew-polyansky) [reply pallets/deip/src/lib.rs]: Typos 👌

**Pull request merged**